src/client/extension.ts

```diff
@@ -57,7 +59,12 @@ export async function activate(
                 !isCurrentDocumentEdited() &&
                 PortalWebView.checkDocumentIsHTML()
             ) {
-                PortalWebView?.currentPanel?._update();
+                if ( PortalWebView?.currentPanel) {
+                    const previewTelemetryData = new TelemetryData(TelemetryConstants.PORTAL_WEBPAGE_PREVIEW);
```

Comment:
why the creation of a new class just to pass its memers into the sendTelemetryEvent? Simply set the properties in the function.
```ts
_telemetry.sendTelemetryEvent('portalWebPagePreview', { preview: 'portalWebPagePreview_NewPage' });
```
this is a ton simpler to read and understand. (note I also removed all the constants which make this bit of code even more harder to understand).